### DISCUSSION TITLE
Unsure how to sign typed data with Web3Modal

### DISCUSSION THREAD

**Discussion by pingu2k4** [WalletConnect/walletconnect-monorepo#2780] [Developer Support]
Hi,

I have Web3Modal and have just bumped to V2. In doing so and testing with the test wallet: https://react-web3wallet.vercel.app/walletconnect, that it doesn't request permission for signing typed data. And indeed, I am not able to do so.

When connecting with metamask browser extension, sign typed data works fine.

I can't see a way to have Web3Modal add sign typed data to the permissions request on connection?

**Comment by xzilja** [comment]
You should use wagmi hook (or associated core action) for these i.e https://wagmi.sh/react/hooks/useSignTypedData
I'm not sure if https://react-web3wallet.vercel.app/walletconnect example supports sign typed data calls (its a simple wallet for reference implementation with walletconnect)

**Reply by pingu2k4** [reply]
Hey.

I'm not using react, I'm using this as plain HTML / JS (Well, its Blazor, but I've got js interop)

I've got the following code to configure Web3Modal (with relevant options being passed down etc), and the only thing necessary to connect after this is calling `openModal()` on the `modal`. The example wallet linked above shows the incoming request as not having requested to be able to sign typed data.

```js
let {projectId, selectedChains, autoConnect, themeOptions, mobileWallets, desktopWallets, walletImages,
       chainImages, tokenImages, explorerAllowList, explorerDenyList, termsOfServiceUrl, privacyPolicyUrl,
       enableNetworkView, enableAccountView, enableExplorer} = JSON.parse(options);

  const convertedChains = convertChains(selectedChains);
  const { provider } = configureChains(convertedChains, [w3mProvider({projectId})]);

  const wagmiClient = createClient({
    autoConnect: autoConnect,
    connectors: w3mConnectors({projectId, version: 2, convertedChains}),
    provider
  });

  ethereumClient = new EthereumClient(wagmiClient, convertedChains);
  modal = new Web3Modal({
    projectId,
    themeMode: themeOptions.themeMode == 0 ? 'light' : 'dark',
    themeVariables: themeOptions.themeVariables,
    mobileWallets: mobileWallets,
    desktopWallets: desktopWallets,
    walletImages: walletImages,
    chainImages: chainImages,
    tokenImages: tokenImages,
    explorerAllowList: explorerAllowList,
    explorerDenyList: explorerDenyList,
    termsOfServiceUrl: termsOfServiceUrl,
    privacyPolicyUrl: privacyPolicyUrl,
    enableNetworkView: enableNetworkView,
    enableAccountView: enableAccountView,
    enableExplorer: enableExplorer
  }, ethereumClient);
```

In my section for signing typed data, this is the code I have - where `signTypedData` is imported from "@wagmi/core":

```js
export async function trySignTypedData(domain, types, value) {
  if(!configured) {
    throw "Attempting to sign typed data before we have configured.";
  }

  try {
    let parsedDomain = JSON.parse(domain);
    let parsedTypes = JSON.parse(types);
    let parsedValue = JSON.parse(value);

    const result = await signTypedData({
      domain: parsedDomain,
      types: parsedTypes.types,
      value: parsedValue
    });

    return getSuccessResponse(result);
  }
  catch(e) {
    return getErrorResponse(e);
  }
}
```

Is it just the case that the example wallet wasn't built to handle signing typed data? If so thats fine I just thought perhaps I should be requesting the ability to call that method, as the interface shows the following, which implies the ability to request permission for more methods:

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/11229848/7d7d9eb2-7ba5-4eca-bd49-43b0b1817216)
